arkane/reference.py

```diff
@@ -0,0 +1,430 @@
+                sources = self.reference_data.keys()
+                data = self.reference_data.values()
+                preferred_source = sources[0]  # If all else fails, use the first source as the preferred one
+                uncertainty = data[0].thermo_data.H298.uncertainty_si
+                for i, entry in enumerate(data):
+                    if (entry.thermo_data.H298.uncertainty_si > 0) and \
+                            (entry.thermo_data.H298.uncertainty_si < uncertainty):
+                        uncertainty = entry.thermo_data.H298.uncertainty_si
+                        preferred_source = sources[i]
+        high_level_h298 = self.reference_data[preferred_source].thermo_data.H298.__reduce__()[1]
+        low_level_h298 = self.calculated_data[model_chemistry].thermo_data.H298.__reduce__()[1]
+
+        return ErrorCancelingSpecies(molecule, low_level_h298, model_chemistry, high_level_h298, preferred_source)
+
+
+class ReferenceDataEntry(RMGObject):
+    """
+    A class for storing reference data for a specific species from a single source
+    """
+    def __init__(self, thermo_data, atct_id=None):
+        """
+
+        Args:
+            thermo_data (rmgpy.thermo.ThermoData): Thermochemistry (Hf298, Cp, ...) from the reference for a species
+            atct_id (str): ID number in the Active Thermochemical Tables if the source is ATcT
+        """
+        super(ReferenceDataEntry, self).__init__()
+        self._thermo_data = None
+        self.thermo_data = thermo_data
+        self.atct_id = atct_id
+
+    def __repr__(self):
+        return str(self.as_dict())
+
+    @property
+    def thermo_data(self):
+        return self._thermo_data
+
+    @thermo_data.setter
+    def thermo_data(self, value):
+        if value:
+            if isinstance(value, ThermoData):
+                self._thermo_data = value
+            else:
+                raise ValueError('thermo_data for a ReferenceDataEntry object must be an rmgpy ThermoData instance')
+        else:
+            self._thermo_data = None
+
+
+class CalculatedDataEntry(RMGObject):
+    """
+    A class for storing a single entry of statistical mechanical and thermochemistry information calculated at a single
+    model chemistry or level of theory
+    """
+    def __init__(self, conformer, thermo, thermo_data, t1_diagnotic=None, fod=None):
+        """
+
+        Args:
+            conformer (rmgpy.statmech.Conformer): Conformer object generated from an Arkane job. Stores many peices of
+                information gained from quantum chemistry calculations, including coordinates, frequencies etc.
+            thermo (HeatCapacityModel): `NASA` or `Wilhoit` thermo object to store the fitted polynomials
+            thermo_data (rmgpy.thermo.ThermoData): Actual thermochemistry values calculated using statistical mechanics
+                at select points. Arkane fits a heat capacity model to this data
```

Comment:
Since `thermo` has a `.toThermoData()` function, perhaps we should make the `thermo_data` argument optional, and if `thermo_data` is passed to CalculatedDataEntry(), ensure that `thermo.toThermoData()` is consistent with `thermo_data`.  Alternatively, maybe we should just get rid of the `thermo_data` argument since we can get `thermo_data` from `thermo`

Reply:
I think the relationship is actually the other way around. ThermoData stores the rigorously correct (up to RRHO) data that we then fit polynomials to. Therefore, I'd rather keep the ThermoData objects since we can derive NASAPolynomials from this data. I have removed the thermo attribute but will add a method to obtain NASAPolynomials from the ThermoData